src/eth/decoder/decoder.ts

```diff
@@ -51,8 +47,15 @@ export class EthDecoder implements Decoder<FunctionCallExplanation> {
   
   protected readonly methodsById: MethodIdMapping;
 
-
-  public decode(data: Buffer): FunctionCallExplanation {
+  public decode(data: Buffer | string): FunctionCallExplanation {
+    if (typeof data === 'string') {
+      data = stripHexPrefix(data);
+      if (isHex(data)) {
+        data = Buffer.from(data, 'hex');
```

Comment:
strip hex prefix and cast to buffer could be wrapped in a function. It looks like both steps are performed together

Reply:
We have created this function, I think is not worth wrap that part. what are your thoughts about? 

`export function parseToBuffer(data: Buffer | string): Buffer {
  if (typeof data === 'string') {
    data = stripHexPrefix(data);
    if (isHex(data)) {
      data = Buffer.from(data, 'hex');
    } else {
      throw new Error('String data is not on hex format');
    }
  }
  return data;
}
`

Reply:
we have chat this offline and we are changing this.